Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -1353,6 +1353,56 @@ class ConstantFoldingPhase : public Phase {
                 break;
             }
 
+            case ArithBitOr: {
+                if (node->child1().useKind() == UntypedUse)
+                    break;
+                if (node->child2()->isInt32Constant() && !node->child2()->asInt32()) {
+                    m_insertionSet.insertCheck(m_graph, indexInBlock, node);
+                    node->convertToIdentityOn(node->child1().node());
+                    changed = true;
+                    break;
```

Comment:
This `break` is redundant.

Reply:
This is kind of early return and we would like to have that so that we can easily place more optimization after this if-clause.